### PR TITLE
Empty universal search

### PR DESCRIPTION
**Pull request by cea2aj**
Support empty universal searches

J=SLAP-1153
TEST=manual, auto

Refer to the minispec and test the 5 listed use cases which are combinations of allowEmptySearch (true or false), defaultInitialSearch (empty string, string, or not supplied). Also modify the data transformer to mock empty universal search results and make sure that the "no results" message shows up in accordance with the spec.

Add a unit test to simulate a default initial search. I added a `wasSearchRanPromise` because the search method is ran asynchronously and if we try to use a standard mock function with expect().toHaveBeenCalled(), the test fails because the expect runs before the search method

### CODE REVIEW

File: src/ui/components/results/universalresultscomponent.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ export default class UniversalResultsComponent extends Component {
       isPreSearch: searchState === SearchStates.PRE_SEARCH,
       isSearchLoading: searchState === SearchStates.SEARCH_LOADING,
       isSearchComplete: searchState === SearchStates.SEARCH_COMPLETE,
-      showNoResults: sections.length === 0 && query,
+      showNoResults: sections.length === 0 && (query || query === ''),
```
Comment (tmeyer2115) [inline src/ui/components/results/universalresultscomponent.js:62]: Do we need to add `allowEmptySearch` in here at all? Like `(query || (query === ''" && allowEmptySearch)`. Or am I worried about an edge case that doesn't exist?

File: tests/ui/components/search/searchcomponent.js [hunk 2]
```diff
@@ -79,4 +80,23 @@ describe('SearchBar component', () => {
...
+    storage.set(StorageKeys.QUERY, defaultInitialSearch);
+    storage.set(StorageKeys.QUERY_TRIGGER, QueryTriggers.INITIALIZE);
+
+    const component = COMPONENT_MANAGER.create('SearchBar', {
+      ...defaultConfig,
+      allowEmptySearch: true
+    });
+
+    const wasSearchRanPromise = new Promise(resolve => {
+      component.search = jest.fn(() => {
```
Comment (oshi97) [inline tests/ui/components/search/searchcomponent.js]: would it make sense to mock core.search instead?

File: src/ui/components/results/universalresultscomponent.js (hunk 1 above)
Comment (cea2aj) [reply src/ui/components/results/universalresultscomponent.js:62]: The search component prevents empty searches from occurring if `allowEmptySearch` is false, so extra logic in the universalresultscomponent isn't necessary

File: tests/ui/components/search/searchcomponent.js (hunk 2 above)
Comment (cea2aj) [reply tests/ui/components/search/searchcomponent.js]: I tried that, but I wasn't able to call the promise resolve function from within this test. Do you have other ideas on how to test that it was called in an asynchronous way?

Comment (oshi97) [reply tests/ui/components/search/searchcomponent.js]: ah I see that IS tricky because there's no callback for the component's async stuff. I think we could still change the mock from component.search to component.core.search though

Comment (cea2aj) [reply tests/ui/components/search/searchcomponent.js]: updated!

**Review by oshi97 — APPROVED** [review]
lgtm!
